sqlx-core/src/postgres/array.rs

```diff
@@ -0,0 +1,21 @@
+use crate::postgres::PgTypeInfo;
+
+pub trait PgHasArrayType {
+    fn array_type_info() -> PgTypeInfo;
+    fn array_compatible(ty: &PgTypeInfo) -> bool {
+        *ty == Self::array_type_info()
+    }
+}
+
+impl<T> PgHasArrayType for Option<T>
+where
+    T: PgHasArrayType,
+{
+    fn array_type_info() -> PgTypeInfo {
+        T::array_type_info()
+    }
+
+    fn array_compatible(ty: &PgTypeInfo) -> bool {
+        T::array_compatible(ty)
+    }
+}
```

Comment:
You mean this function, or why there is a new `array` module at all?

Reply:
Why the new module, yeah.

Reply:
In a way this code seemed more "fundamental" than what's inside `types/` since it introduces another trait re-exported at `sqlx::postgres` rather than just implementing traits from higher up the module tree.

Happy to move things around though.

Reply:
> Happy to move things around though.

@abonander should I?

Reply:
Yeah, I would prefer this to be merged into `postgres/types/array.rs`.